library/core/src/ptr/mod.rs

```diff
@@ -450,7 +450,7 @@ mod mut_ptr;
 
 /// Executes the destructor (if any) of the pointed-to value.
 ///
-/// This is semantically equivalent to calling [`ptr::read`] and discarding
+/// This is almost the same as calling [`ptr::read`] and discarding
 /// the result, but has the following advantages:
```

Comment:
Could this be something like

```suggestion
/// For most cases, this is equivalent to calling [`ptr::read`] and discarding
/// the result, but it is different in the following ways:
```

or is there a reason we should not prefer that more precise statement?

Reply:
See the discussion in https://github.com/rust-lang/rust/issues/112015 -- it's not fully clear whether we really want it to be equivalent, e.g. when calling `read` on a `*mut &T` we definitely require the reference to be dereferenceable, but when calling `drop_in_place` maybe we don't.

Reply:
somewhere around line 540, I think.

Reply:
I added a FIXME and a Miri test.

Reply:
Thanks! Digging deep into histories is inevitable if someone wants full context, but it's nice to be able to see a plain-text hint in the source.